### PR TITLE
Added configurable 'player_save_interval' seconds property

### PR DESCRIPTION
**Pull request by dgarson**
Adds a new configuration property for the number of minutes between the automatic save of player characters. This is in response to the the feature request in https://github.com/ACEmulator/ACE/issues/1621

### CODE REVIEW

File: Source/ACE.Server/Managers/PlayerManager.cs [hunk 1]
```diff
@@ -27,11 +27,18 @@ public static class PlayerManager
         private static readonly Dictionary<uint, Player> onlinePlayers = new Dictionary<uint, Player>();
         private static readonly Dictionary<uint, OfflinePlayer> offlinePlayers = new Dictionary<uint, OfflinePlayer>();
 
+        public static readonly long DefaultPlayerSaveInterval = 5;
```
Comment (Mag-nus) [inline Source/ACE.Server/Managers/PlayerManager.cs]: When I was thinking about this, I was thinking the unit of measurement would be seconds.

File: Source/ACE.Server/WorldObjects/Player.cs [hunk 2]
```diff
@@ -47,6 +47,11 @@ public partial class Player : Creature, IPlayer
 
         public float CurrentRadarRange => Location.Indoors ? 25.0f : 75.0f;
 
+        public TimeSpan PlayerSaveInterval
```
Comment (Mag-nus) [inline Source/ACE.Server/WorldObjects/Player.cs]: This can go back in Player_Database.cs

File: Source/ACE.Server/Managers/PropertyManager.cs [hunk 3]
```diff
@@ -469,8 +475,9 @@ public static void LoadDefaultProperties()
                 ("char_delete_time", 3600),         // the amount of time in seconds a deleted character can be restored
                 ("mansion_min_rank", 6),            // overrides the default allegiance rank required to own a mansion
                 ("max_chars_per_account", 11),      // retail defaults to 11, client supports up to 20
-                ("pk_timer", 20)                    // the number of seconds where a player cannot perform certain actions (ie. teleporting)
+                ("pk_timer", 20),                   // the number of seconds where a player cannot perform certain actions (ie. teleporting)
                                                     // after becoming involved in a PK battle
+                ("player_save_interval", 5)         // number of minutes between saving of characters
```
Comment (Mag-nus) [inline Source/ACE.Server/Managers/PropertyManager.cs]: Seconds make sense here. It's the unit of measurement we use in the majority (all?) of cases in ACE

File: Source/ACE.Server/Managers/PlayerManager.cs (hunk 1 above)
Comment (dgarson) [reply Source/ACE.Server/Managers/PlayerManager.cs]: I'll add a suffix indicating it's minutes.

File: Source/ACE.Server/WorldObjects/Player.cs (hunk 2 above)
Comment (dgarson) [reply Source/ACE.Server/WorldObjects/Player.cs]: Woops. This isn't supposed to be in here anymore. Thanks for catching that.

File: Source/ACE.Server/Managers/PropertyManager.cs (hunk 3 above)
Comment (dgarson) [reply Source/ACE.Server/Managers/PropertyManager.cs]: I was just using the existing units which was minutes. Seconds works too. I'll switch it over.

File: Source/ACE.Server/WorldObjects/Player.cs (hunk 2 above)
Comment (dgarson) [reply Source/ACE.Server/WorldObjects/Player.cs]: I remember what happened here. It says `PlayerManager` is not in scope for Player_Database

**Comment by Mag-nus** [comment]
Thanks for the seconds change.

My only other thought is that PlayerManager should not be involved.

I think you can just have PlayerSaveInterval in Player_Dabase.cs point to => PropertyManager.GetLong("player_save_interval", DefaultPlayerSaveIntervalSecs).Item;

You can change PlayerSaveInterval to an int so you're not converting every request.

If we want to capture changes from Propertymanager, it should raise an event that also includes the property being changed. Then, objects could subscribe to that event to determine if they need to make an adjustment.

**Comment by dgarson** [comment]
I found that PropertyManager was also inaccessible from Player_Database either. Any other ideas?

I agree about the PropertyManager change callback but that felt outside the scope of this PR.

**Comment by Theran578** [comment]
Adding 'using ACE.Server.Managers;' should make PropertyManager "accessible" from Player_Database.cs.  As far as I can see, the relevant API functions of PropertyManager are all set to public.

**Comment by dgarson** [comment]
@Mag-nus any more changes I should make to this PR before it's ready to merge?

**Pull request merged**